### ISSUE TITLE
Mail without run information, when there was spot termination

### ISSUE THREAD

**Issue by fruch** [scylladb/scylla-cluster-tests#4656]
On master run while there was spot termination, mail data wasn't collected cause of the following reason,
I would expect those collection to continue regardless of errors (each one of them, and not fail the saving of data)

```
[2022-04-28T05:04:14.643Z] 2022-04-28 05:04:01.444: (TestFrameworkEvent Severity.ERROR), source=LongevityTest.save_email_data() message=save_email_data (silenced) failed with:
[2022-04-28T05:04:14.643Z] exception=Unable to run "nproc": failed connecting to "10.0.1.113" during 60s
[2022-04-28T05:04:14.643Z] Traceback (most recent call last):
[2022-04-28T05:04:14.643Z] File "/home/ubuntu/scylla-cluster-tests/./sct.py", line 908, in <module>
[2022-04-28T05:04:14.643Z] cli()
[2022-04-28T05:04:14.643Z] File "/usr/local/lib/python3.9/site-packages/click/core.py", line 764, in __call__
[2022-04-28T05:04:14.643Z] return self.main(*args, **kwargs)
[2022-04-28T05:04:14.643Z] File "/usr/local/lib/python3.9/site-packages/click/core.py", line 717, in main
[2022-04-28T05:04:14.643Z] rv = self.invoke(ctx)
[2022-04-28T05:04:14.643Z] File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1137, in invoke
[2022-04-28T05:04:14.643Z] return _process_result(sub_ctx.command.invoke(sub_ctx))
[2022-04-28T05:04:14.643Z] File "/usr/local/lib/python3.9/site-packages/click/core.py", line 956, in invoke
[2022-04-28T05:04:14.643Z] return ctx.invoke(self.callback, **ctx.params)
[2022-04-28T05:04:14.643Z] File "/usr/local/lib/python3.9/site-packages/click/core.py", line 555, in invoke
[2022-04-28T05:04:14.643Z] return callback(*args, **kwargs)
[2022-04-28T05:04:14.643Z] File "/home/ubuntu/scylla-cluster-tests/./sct.py", line 658, in run_test
[2022-04-28T05:04:14.643Z] unittest.main(module=None, argv=['python -m unittest', argv],
[2022-04-28T05:04:14.643Z] File "/usr/local/lib/python3.9/unittest/main.py", line 101, in __init__
[2022-04-28T05:04:14.643Z] self.runTests()
[2022-04-28T05:04:14.643Z] File "/usr/local/lib/python3.9/unittest/main.py", line 271, in runTests
[2022-04-28T05:04:14.643Z] self.result = testRunner.run(self.test)
[2022-04-28T05:04:14.643Z] File "/usr/local/lib/python3.9/unittest/runner.py", line 176, in run
[2022-04-28T05:04:14.643Z] test(result)
[2022-04-28T05:04:14.643Z] File "/usr/local/lib/python3.9/unittest/suite.py", line 84, in __call__
[2022-04-28T05:04:14.643Z] return self.run(*args, **kwds)
[2022-04-28T05:04:14.643Z] File "/usr/local/lib/python3.9/unittest/suite.py", line 122, in run
[2022-04-28T05:04:14.643Z] test(result)
[2022-04-28T05:04:14.643Z] File "/usr/local/lib/python3.9/unittest/suite.py", line 84, in __call__
[2022-04-28T05:04:14.643Z] return self.run(*args, **kwds)
[2022-04-28T05:04:14.643Z] File "/usr/local/lib/python3.9/unittest/suite.py", line 122, in run
[2022-04-28T05:04:14.643Z] test(result)
[2022-04-28T05:04:14.643Z] File "/usr/local/lib/python3.9/unittest/case.py", line 653, in __call__
[2022-04-28T05:04:14.643Z] return self.run(*args, **kwds)
[2022-04-28T05:04:14.643Z] File "/home/ubuntu/scylla-cluster-tests/sdcm/tester.py", line 389, in run
[2022-04-28T05:04:14.643Z] result = super(ClusterTester, self).run(self.result)
[2022-04-28T05:04:14.643Z] File "/usr/local/lib/python3.9/unittest/case.py", line 596, in run
[2022-04-28T05:04:14.643Z] self._callTearDown()
[2022-04-28T05:04:14.643Z] File "/usr/local/lib/python3.9/unittest/case.py", line 553, in _callTearDown
[2022-04-28T05:04:14.643Z] self.tearDown()
[2022-04-28T05:04:14.643Z] File "/home/ubuntu/scylla-cluster-tests/sdcm/tester.py", line 2034, in tearDown
[2022-04-28T05:04:14.643Z] self.save_email_data()
[2022-04-28T05:04:14.643Z] File "/home/ubuntu/scylla-cluster-tests/sdcm/tester.py", line 177, in decor
[2022-04-28T05:04:14.643Z] self._store_test_result(args[0], exc, exc.__traceback__, name)
```

CI run:
https://jenkins.scylladb.com/job/enterprise-2021.1/job/longevity/job/longevity-alternator-3h-test/91/

**Comment by fruch** [comment]
was fixed in https://github.com/scylladb/scylla-cluster-tests/pull/3736
but never got backported to 2021.1, just to 4.5

**Comment by fruch** [comment]
backported the needed fix to 2021.1
